app/models/manageiq/providers/lenovo/physical_infra_manager/event_catcher/stream.rb

```diff
@@ -27,12 +27,19 @@ def filter_fields
       { :operation => 'NOT', :field => 'eventClass', :value => '200' },
       { :operation => 'NOT', :field => 'eventClass', :value => '800' }
     ]
-    last_cn_event = get_last_cnn_from_events(@ems.id)
+    last_cn_event = get_last_ems_ref(@ems.id)
     cn_operation = { :operation => 'GT', :field => 'cn', :value => last_cn_event.to_s }
     fields.push(cn_operation) unless last_cn_event.nil?
     fields
   end
 
+  def parse_events(events)
+    events.collect do |data|
+      event = ManageIQ::Providers::Lenovo::PhysicalInfraManager::EventCatcher::Event.new(data).to_hash
+      ManageIQ::Providers::Lenovo::PhysicalInfraManager::EventParser.event_to_hash(event, @ems.id)
```

Comment:
What is the point of splitting these two up?  The whole point of the `EventParser` is to parse the raw event into a hash so I don't see the utility in adding `ManageIQ::Providers::Lenovo::PhysicalInfraManager::EventCatcher::Event`.